Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/BaseInteractiveElement.cs

```diff
@@ -0,0 +1,247 @@
+        /// </summary>
+        /// <param name="coreState">The CoreInteractionState to retrieve</param>
+        /// <returns>The Core Interaction State contained in the Tracked States scriptable object.</returns>
+        public InteractionState GetState(CoreInteractionState coreState)
+        {
+            return StateManager.GetState(coreState);
+        }
+
+        /// <summary>
+        /// Gets a state by using the state name.
+        /// </summary>
+        /// <param name="stateName">The name of the state to retrieve</param>
+        /// <returns>The state contained in the Tracked States scriptable object.</returns>
+        public InteractionState GetState(string stateName)
+        {
+            return StateManager.GetState(stateName);
+        }
+
+        /// <summary>
+        /// Adds a new Core Interaction State to track.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to add</param>
+        /// <returns>The newly added Core Interaction State</returns>
+        public InteractionState AddCoreState(CoreInteractionState state)
+        {
+            return StateManager.AddCoreState(state);
+        }
+
+        /// <summary>
+        /// Creates and adds a new state to track given the new state name.
+        /// </summary>
+        /// <param name="stateName">The name of the state to add</param>
+        /// <returns>The new state added</returns>
+        public InteractionState AddNewState(string stateName)
+        {
+            return StateManager.AddNewState(stateName);
+        }
+
+        /// <summary>
+        /// Removes a Core Interaction State. The state will no longer be tracked if it is removed.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to remove</param>
+        public void RemoveCoreState(CoreInteractionState state)
+        {
+            StateManager.RemoveState(state);
+        }
+
+        /// <summary>
+        /// Removes a state. The state will no longer be tracked if it is removed.
+        /// </summary>
+        /// <param name="stateName">The name of the state to remove</param>
+        public void RemoveState(string stateName)
+        {
+            StateManager.RemoveState(stateName);
+        }
+
+        /// <summary>
+        /// Create and add a new state given the state name and the associated existing event configuration.
+        /// </summary>
+        /// <param name="stateName">The name of the state to create</param>
+        /// <param name="eventConfiguration">The existing event configuration for the new state</param>
+        /// <returns>The new state added</returns>
+        public void AddNewStateWithEventConfiguration(string stateName, BaseInteractionEventConfiguration eventConfiguration)
+        {
+            StateManager.AddNewStateWithEventConfiguration(stateName, eventConfiguration);
+        }
+
+        /// <summary>
+        /// Check if a state is currently being tracked.
```

Comment:
What does "tracked" mean?

Reply:
Tracked means that the state has been added to the state manager.  For example, here the focus state is being tracked by the state manager:

![image](https://user-images.githubusercontent.com/53493796/93107388-63390600-f666-11ea-9340-d2931576fdc0.png)

Here the focus state is not being tracked by the state manager:

![image](https://user-images.githubusercontent.com/53493796/93107628-b6ab5400-f666-11ea-8a97-07d839a2967b.png)

I added the clarification in the comment:

```
Checks if a state is currently being tracked by the state manager.
```